File: functions/libsodium/variants/c/1_b64_byte_to_urlsafe_char_9.c
#include <stdint.h>

static int b64_byte_to_urlsafe_char(unsigned int x)
{
    if (x <= 25) return x + 'A';
    if (x <= 51) return x - 26 + 'a';
    if (x <= 61) return x - 52 + '0';
    if (x == 62) return '-';
    return '_';
}


int main(){
    b64_byte_to_urlsafe_char(0);
}

File: functions/libsodium/variants/go/1_b64_byte_to_urlsafe_char_9.go
package main

func b64ByteToURLSafeChar(x uint32) int32 {
    switch {
    case x >= 0 && x < 26:
        return int32(x) + 'A'
    case x >= 26 && x < 52:
        return int32(x-26) + 'a'
    case x >= 52 && x < 62:
        return int32(x-52) + '0'
    case x == 62:
        return '-'
    case x == 63:
        return '_'
    default:
        return -1
    }
}


func main(){
    b64ByteToURLSafeChar(0)
}
